ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestCasesRunner.scala

```diff
@@ -140,65 +148,83 @@ final class LedgerTestCasesRunner(
+  ): Future[Vector[LedgerTestSummary]] = {
+    val testCount = testCases.size
+    logger.info(s"Running $testCount tests, ${math.min(testCount, concurrency)} at a time.")
+    Source(testCases.zipWithIndex)
+      .mapAsyncUnordered(concurrency) { case (test, index) =>
+        run(test, ledgerSession).map(summarize(test.suite, test, _) -> index)
+      }
+      .runWith(Sink.seq)
+      .map(_.toVector.sortBy(_._2).map(_._1))
+  }
 
+  private def run(
+      participants: Vector[Channel]
+  )(implicit
+      materializer: Materializer,
+      executionContext: ExecutionContext,
+  ): Future[Vector[LedgerTestSummary]] = {
     val (concurrentTestCases, sequentialTestCases) = testCases.partition(_.runConcurrently)
-    ParticipantSessionManager(config.participants)
-      .flatMap { participantSessionManager =>
-        val ledgerSession = LedgerSession(participantSessionManager, config.shuffleParticipants)
+    ParticipantSession(partyAllocation, participants, commandInterceptors)
+      .flatMap { sessions =>
+        val ledgerSession = LedgerSession(sessions, shuffleParticipants)
         val testResults =
           for {
-            _ <- uploadDarsIfRequired(participantSessionManager)
+            _ <- uploadDarsIfRequired(sessions)
             concurrentTestResults <- runTestCases(
               ledgerSession,
               concurrentTestCases,
               concurrentTestRuns,
-            )
+            )(materializer, materializer.executionContext)
             sequentialTestResults <- runTestCases(
               ledgerSession,
               sequentialTestCases,
               concurrency = 1,
-            )
+            )(materializer, materializer.executionContext)
           } yield concurrentTestResults ++ sequentialTestResults
 
         testResults
-          .recover { case NonFatal(e) => throw new LedgerTestCasesRunner.UncaughtExceptionError(e) }
-          .andThen { case _ => participantSessionManager.disconnectAll() }
-      }
-      .andThen { case _ =>
-        materializer.shutdown()
-        system.terminate().failed.foreach { throwable =>
-          logger.error("The actor system failed to terminate.", throwable)
-        }
-      }
-      .onComplete { result =>
-        completionCallback(result)
+          .recover { case NonFatal(e) =>
+            throw new LedgerTestCasesRunner.UncaughtExceptionError(e)
+          }
       }
   }
 
-  def verifyRequirementsAndRun(completionCallback: Try[Vector[LedgerTestSummary]] => Unit): Unit = {
+  private def prepareResourcesAndRun(implicit
+      executionContext: ExecutionContext
+  ): Future[Vector[LedgerTestSummary]] = {
+
+    val materializerResources =
+      ResourceOwner.forMaterializerDirectly(createActorSystem).acquire()
+
+    // When running the tests, explicitly use the materializer's execution context
+    // The tests will then be executed on it instead of the implicit one -- which
+    // should only be used to
```

Comment:
Seems like there is something missing in this sentence?

Reply:
It's definitely not missing suspense.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/8607/commits/ef73c3d4fe8571375406332aa82dc0f2d42496da